### PR TITLE
Add support for ReadonlyMap and ReadonlySet

### PR DESCRIPTION
**Pull request by mikhailswift**

### CODE REVIEW

File: lib/types.ts [hunk 1]
```diff
@@ -97,10 +113,14 @@ export type DeepWritable<T> = T extends Builtin
   ? T
   : T extends Map<infer K, infer V>
   ? Map<DeepWritable<K>, DeepWritable<V>>
+  : T extends ReadonlyMap<infer K, infer V>
```
Comment (krzkaczor) [inline lib/types.ts]: Shouldn't it become just `Map` for writable types? These `Readonly*` types require special care IMO...

@quezak WDYT?

Comment (quezak) [reply lib/types.ts]: yeah, I think it should -- writable is writable

Comment (quezak) [reply lib/types.ts]: but, if we do this, `DeepReadonly` for a regular `Map` should also return `ReadonlyMap<K, DeepReadonly<V>>`, for consistency

Comment (mikhailswift) [reply lib/types.ts]: That all makes sense.  Made those changes.

**Pull request merged**